app/models/manageiq/providers/infra_manager.rb

```diff
@@ -7,6 +7,18 @@ class InfraManager < BaseManager
 
     include AvailabilityMixin
 
+    has_many :host_hardwares,         :through => :hosts, :source => :hardware
+    has_many :host_operating_systems, :through => :hosts, :source => :operating_system
+    has_many :host_storages,          :through => :hosts
+    has_many :host_switches,          :through => :hosts
+    has_many :host_networks,          :through => :hosts, :source => :networks
+    has_many :snapshots,              :through => :vms_and_templates
+    has_many :operating_systems,      :through => :vms_and_templates
+    has_many :switches,               :through => :hardwares
+    has_many :lans,                   :through => :hardwares
```

Comment:
In the host we could access lans and switches through host_switches which seems to be more complex query then when using [1]. Do you want me to change it?

[1] https://github.com/ManageIQ/manageiq/pull/15441/files#diff-6d975a235a345bb9f0d3375e70aaf691R23